File: app/Models/Restaurant.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Restaurant extends Model
{
    protected $fillable = [
        'owner_id',
        'name',
        'description',
        'phone',
        'address',
        'latitude',
        'longitude',
        'delivery_fee',
        'minimum_order',
        'image_url',
        'is_open',
        'is_active',
    ];

    protected function casts(): array
    {
        return [
            'latitude' => 'decimal:7',
            'longitude' => 'decimal:7',
            'delivery_fee' => 'decimal:2',
            'minimum_order' => 'decimal:2',
            'is_open' => 'boolean',
            'is_active' => 'boolean',
        ];
    }

    public function owner(): BelongsTo
    {
        return $this->belongsTo(User::class, 'owner_id');
    }

    public function categories(): HasMany
    {
        return $this->hasMany(Category::class);
    }

    public function orders(): HasMany
    {
        return $this->hasMany(Order::class);
    }


}

File: mobile/lib/features/restaurants/domain/restaurant.dart
import 'category.dart';

class Restaurant {
  const Restaurant({
    required this.id,
    required this.name,
    required this.description,
    required this.phone,
    required this.address,
    required this.deliveryFee,
    required this.minimumOrder,
    required this.imageUrl,
    required this.isOpen,
    this.categories = const [],
  });

  final int id;
  final String name;
  final String? description;
  final String? phone;
  final String address;
  final double deliveryFee;
  final double minimumOrder;
  final String? imageUrl;
  final bool isOpen;
  final List<Category> categories;

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    final categories = json['categories'] as List<dynamic>? ?? [];

    return Restaurant(
      id: json['id'] as int,
      name: json['name'] as String,
      description: json['description'] as String?,
      phone: json['phone'] as String?,
      address: json['address'] as String,
      deliveryFee: double.parse(json['delivery_fee'].toString()),
      minimumOrder: double.parse(json['minimum_order'].toString()),
      imageUrl: json['image_url'] as String?,
      isOpen: json['is_open'] as bool,
      categories: categories
          .map(
            (item) =>
                Category.fromJson(Map<String, dynamic>.from(item as Map)),
          )
          .toList(),
    );
  }
}
